Add memcached recipe

radamanthus opened the pull request:
## Description of your patch

Adds the memcached custom recipe. 

The previous recipe was just a stub. This actually installs memcahced, but it has to be enabled by setting `perform_install` to true in `custom-memcached/attributes/default.rb`.

## Recommended Release Notes

Adds the memcached custom recipe.

## Estimated risk

High. 

Our platform includes memcached in the list of recipes that should be run by default (a legacy from the V4 stack). Breakage on this recipe will cause main chef to fail for all chef runs on V5. To mitigate this, the recipe has an attribute `perform_install` and this is disabled by default.

## Components involved

memcached custom chef recipe

## Description of testing done

See QA instructions

## QA Instructions

Run the test cases in examples/redis/cookbooks/custom-memcached/README.md

dvalfre commented on cookbooks/memcached/README.md:
```diff
@@ -1,4 +1,11 @@
+    run "ln -nfs #{shared_path}/config/memcached_custom.yml #{release_path}/config/memcached.yml"
+    
+
+[EY Cloud]: https://cloud.engineyard.com/extras
```
This URL gives a 404.

dvalfre commented on examples/memcached/cookbooks/custom-memcached/README.md:
```diff
@@ -0,0 +1,121 @@
+A. Do not enable the perform_install flag
+  A1. Chef run should not fail
+  A2. memcached should not be running
+B. Install from source 
```
the `perform_install` flag should be set to true

dvalfre commented on examples/memcached/cookbooks/custom-memcached/README.md:
```diff
@@ -0,0 +1,121 @@
+  B2.1. memcached should be running on the utility instance named memcached
+  B2.2. memcached should not be running on app_master, app, and database instances
+  B2.3. /data/app_name/shared/config/memcached.yml should list all the utility instances named memcached in the environment
+C. Install from package
```
the `perform_install` should be set to `true`

dvalfre commented on examples/memcached/cookbooks/custom-memcached/README.md:
```diff
@@ -0,0 +1,121 @@
+
+If you do not have `cookbooks/ey-custom` on your app repository, you can copy
+`examples/memcached/cookbooks/ey-custom` to `/path/to/app/cookbooks` as well.
+
```
At some point the user should switch `perform_install` to `true`.  It is not necessarily a customization, so not belonging to that section of the Readme.

radamanthus commented:
I updated stuff, squashed the commits, and did a `push --force`. Here's a summary of the changes:

1) Removed the dead link in memcached/README.md

2) Updated the Installation section in examples/memcached/cookbooks/custom-memcached/README.md based on the feedback

3) examples/memcached/cookbooks/custom-memcached/attributes/default.rb - perform_install is set to true. If the customer is adding custom-memcached, we assume he plans to install it :)
